metrics-api/src/main/java/software/amazon/swage/metrics/record/MetricRecorder.java

```diff
@@ -121,6 +123,8 @@ public TypedMap attributes() {
     public MetricContext context(TypedMap attributes) {
         R context = newRecorderContext(attributes);
         return new MetricContext() {
+            private Map<Metric,Long> counts = new ConcurrentHashMap<>();
```

Comment:
See below - I don't believe this belongs here.  Aside from the separation of concerns thing, there is now a map created and stored for every metric context, even if said context never uses it.